addons/account/populate/account_move.py

```diff
@@ -96,19 +96,16 @@ def get_lines(random, values, **kwargs):
             :return list: list of ORM create commands for the field line_ids
             """
             def get_line(account, label, balance=None, balance_sign=False, exclude_from_invoice_tab=False):
-                is_foreign_currency = (
-                    currency_id != account.currency_id.id
-                    and currency_id != account.company_id.currency_id.id
-                )
+                is_foreign_currency = currency_id != account.company_id.currency_id.id
                 balance = balance or balance_sign * round(random.uniform(0, 1000))
                 return (0, 0, {
                     'name': 'label_%s' % label,
                     'debit': balance > 0 and balance or 0,
                     'credit': balance < 0 and -balance or 0,
                     'account_id': account.id,
                     'partner_id': partner_id,
-                    'currency_id': currency_id if is_foreign_currency else False,
-                    'amount_currency': 2 * balance if is_foreign_currency else False,
+                    'currency_id': currency_id,
+                    'amount_currency': 2 * balance if is_foreign_currency else balance,
```

Comment:
We should use the rates calculated in the populate too. It was added later in the PR and I forgot to change this here https://github.com/odoo/odoo/commit/3afbd1f58dd627442cf08abaadb777501521afb0

Reply:
Done :)